### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Yuliya312**
[DEMO LINK](https://Yuliya312.github.io/react_todo-app/)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,85 +1,149 @@
...
+    const { todos, filterButtonsChosed } = this.state;
+    const filteredTodos = this.filterTodos();
+    const checkComplete = todos.every(todo => todo.completed);
+
+    return (
+      <section className="todoapp">
+        <header className="header">
+          <h1>todos</h1>
+          <NewTodo addTodo={this.addTodo} />
+
```
Comment (irenhh) [inline src/App.js]: remove this line

File: src/components/NewTodo/NewTodo.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  changeHandler = (event) => {
+    this.setState({
+      title: event.target.value,
+    });
+  }
+
+  enterHandler = (event) => {
+    const { title } = this.state;
+
+    if (event.key === 'Enter' && title.trim().length >= 1) {
```
Comment (irenhh) [inline src/components/NewTodo/NewTodo.js]: add form so that you don't need to find a key name

File: src/components/TodoItem/TodoItem.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      title,
+      completed,
+    },
+    onCheckboxChecked,
+    onTodoDelete,
+  } = props;
+
+  return (
+    <li
+      className={completed ? 'completed' : ''}
```
Comment (irenhh) [inline src/components/TodoItem/TodoItem.js]: use classnames lib

File: src/components/NewTodo/NewTodo.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import { v4 as uuidv4 } from 'uuid';
+
+export class NewTodo extends Component {
+  static propTypes = {
```
Comment (irenhh) [inline src/components/NewTodo/NewTodo.js]: be consistent about where you write propTypes

File: src/components/TodoList/TodoList.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { TodoItem } from '../TodoItem/TodoItem';
+
+export const TodoList = (props) => {
+  const { todos, deleteTodo, checkedTodo } = props;
+
+  return (
+    <>
```
Comment (irenhh) [inline src/components/TodoList/TodoList.js:10]: remove this

File: src/components/TodosFilter/TodosFilter.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+                href="#/"
+                className={
+                  filterButtonsChosed === filterButton ? 'selected' : ''
+                }
+                onClick={() => filterHandler(filterButton)}
+              >
+                {filterButton}
+              </a>
+            </li>
+          ))}
```
Comment (irenhh) [inline src/components/TodosFilter/TodosFilter.js:40]: you should close the bracket on the same level you opened it

**Review by irenhh — CHANGES_REQUESTED** [review]
image
it should be active ONLY if there is any todo and all of them are checked

add cursor pointer to all interactive elements

File: src/App.js [hunk 7]
```diff
@@ -1,85 +1,147 @@
...
+    const { filterButtonsChosed, todos } = this.state;
+
+    switch (filterButtonsChosed) {
+      case filterButtons.active:
+        return todos.filter(todo => !todo.completed);
+      case filterButtons.completed:
+        return todos.filter(todo => todo.completed);
+      case filterButtons.all:
+        return [...todos];
+      default:
```
Comment (danheim) [inline src/App.js:61]: комбинируй
```suggestion
      case filterButtons.all:
      default:
```

File: src/App.js [hunk 8]
```diff
@@ -1,85 +1,147 @@
...
+  };
+
+  filterHandler = (filterButton) => {
+    this.setState({
+      filterButtonsChosed: filterButton,
+    });
+  }
+
+  filterTodos = () => {
+    const { filterButtonsChosed, todos } = this.state;
```
Comment (danheim) [inline src/App.js:52]:
```suggestion
    const { filterField, todos } = this.state;
```

File: src/components/TodoItem/TodoItem.js [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+export const TodoItem = (props) => {
+  const {
+    todo: {
+      id,
+      title,
+      completed,
+    },
+    onCheckboxChecked,
+    onTodoDelete,
+  } = props;
```
Comment (danheim) [inline src/components/TodoItem/TodoItem.js:14]: не деструктурируй вложенно, делай это отдельно

File: src/components/TodoList/TodoList.js (hunk 5 above)
Comment (danheim) [reply src/components/TodoList/TodoList.js:10]: remove

**Review by danheim — APPROVED** [review]

**Pull request merged**